docs/source/writing_testbenches.rst

```diff
@@ -247,17 +247,23 @@ the various actions described in :ref:`assignment-methods` can be used.
 
 .. _writing_tbs_accessing_underscore_identifiers:
 
-Accessing Identifiers Starting with an Underscore
-=================================================
+Accessing Identifiers Starting with an Underscore or Invalid Python Names
+=========================================================================
 
 The attribute syntax of ``dut._some_signal`` cannot be used to access
 an identifier that starts with an underscore (``_``, as is valid in Verilog)
 because we reserve such names for cocotb-internals,
 thus the access will raise an :exc:`AttributeError`.
 
-A workaround is to use indirect access using
-:meth:`~cocotb.handle.HierarchyObject._id` like in the following example:
-``dut._id("_some_signal", extended=False)``.
+Both SystemVerilog and VHDL allow developers to create signals or nets with non-standard characters by using special syntax.
+These objects are generally not accessible using attribute syntax since attributes in Python must follow a strict form.
+
+A workaround for both cases is to use index syntax.
+
+.. code-block:: python3
+
+    dut["_some_signal"]  # begins with underscore
+    dut["\\!WOW!\\"]     # uses extended identifiers
```

Comment:
```suggestion
    dut["\\!WOOOOW!\\"]  # escaped identifier (Verilog), extended identifier (VHDL)
```
Make the phrase findable if someone searches for "escaped identifier" with quotes.

Reply:
Also updated the docstring.